botocore/utils.py

```diff
@@ -92,7 +93,9 @@
 # These are chars that do not need to be urlencoded.
 # Based on rfc2986, section 2.3
 SAFE_CHARS = '-._~'
-LABEL_RE = re.compile(r'[a-z0-9][a-z0-9\-]*[a-z0-9]')
+VALID_HOST_LABEL_RE = re.compile(
+    r"^((?!-)[a-zA-Z\d-]{1,63}(?<!-))*((?!-)[a-zA-Z\d-]{1,63}(?<!-))$",
+)
```

Comment:
This regex is not correct because it allows arbitrarily long strings. Assuming that the intent is to enforce the [3,63] limits, and that you want to use the lookback/lookahead technique, you could write it as:

```
^((?!-)[a-zA-Z\d-]{1,63}(?<!-))$
```

However, this simpler one does the same job, as far as I can tell:

```
^[a-zA-Z0-9][a-zA-Z0-9\-]{1,61}[a-zA-Z0-9]$
```

Reply:
ah yeah you're right! 🤦 

I'm a little confused by your comments though. You first say [3,63], then [1,63] then [1,61]

I don't think we should use the lookahead since we can just say only alphanumeric characters, then alphanumeric plus a hyphen from 1-63 characters, then just alphnanumeric. So basically what you have except replace `{1,61}` with `{1,63}` which I'm assuming was just a typo?

Reply:
So actually upon further investigation the second regex won't work because it expects at least 3 characters. The negative look ahead/back is our only option I think

Reply:
Apologies for causing confusion. The current bucket name rules say that bucket names must be between 3 and 63 characters. I have since learned, that there may exists shorter bucket names that have been grandfathered in. In my comment, I tried to be consistent with the current rules, which translate to `[starting character][1-61 middle characters][final character]`. Looks like I messed up and put a 1 where there should have been a 3 or a 63 where there should have been 61, not sure which one.

Reply:
oh no worries i thought it had to be 1-63 characters. i'll push up an update